File: CSharpFundamentals/CSharpOOPbasics/InterfacesAndAbstractionEx/CollectionHierarchy/AddRemoveCollection.cs
using System.Collections.Generic;

namespace CollectionHierarchy
{
    class AddRemoveCollection : IAdd, IRemove
    {
        private List<string> data;

        public AddRemoveCollection()
        {
            this.data = new List<string>();
        }

        public int Add(string element)
        {
            this.data.Insert(0, element);
            return this.data.IndexOf(element);
        }

        public string Remove()
        {
            string element = data[data.Count - 1];
            this.data.RemoveAt(this.data.Count - 1);
            return element;
        }
    }
}

File: JavaFundamentals/JavaOOP/Interfaces-And-Abstraction-Ex/src/collection_hierarchy/AddRemoveCollection.java
package collection_hierarchy;

public class AddRemoveCollection extends Collection implements AddRemovable {
    @Override
    public String remove() {
        return this.getItems().remove(this.getItems().size() - 1);
    }

    @Override
    public int add(String element) {
        this.getItems().add(0, element);
        return 0;
    }
}
